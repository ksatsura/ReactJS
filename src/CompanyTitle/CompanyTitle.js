// import
import React from 'react';
import '../style.css';

const CompanyTitle = () => {

    return <p className='company-title'>netflixroulette</p>;
};

export default CompanyTitle;