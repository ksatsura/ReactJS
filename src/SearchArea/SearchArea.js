//imports
import React from 'react';
import SearchFilter from '../SearchFilter/SearchFilter';
import { connect } from 'react-redux';
import { saveInputValue } from '../actions/searchActions';
import '../style.css';

export const SearchArea = (props) => {

    const { value, handleChange, match, location } = props;

    const paramValue = match && match.params.value;

    return (
        <div className='search-area'>
            <h1 className='search-title'>find your movie</h1>
            <input type="search" name="search-value" className='search-value' value={value} onChange={(e) => handleChange(e.target.value)} />
            <SearchFilter location={location} paramValue={paramValue} />
        </div>
    );  
};

const mapDispatchToProps = dispatch => {
    return {
        handleChange: (value) => dispatch(saveInputValue(value)),
    };
};

const mapStateToProps = state => {
    return {
        value: state.filmListReducer.currentValue,
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchArea);