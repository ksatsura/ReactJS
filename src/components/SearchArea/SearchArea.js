//imports
import * as React from 'react';
import SearchFilter from '../SearchFilter/SearchFilter';
import { connect } from 'react-redux';
import { saveInputValue } from '../../actions/searchActions';
import injectSheet from 'react-jss';
import '../../style.css';

// @flow
type Props = {
    value: string, 
    handleChange: (string) => void, 
    match: {
        params: {
            value: string
        }
    },
    location: {},
};

const styles = {
    searchArea: {
        display: 'flex',
        flexDirection: 'column',
    },
    searchTitle: {
        color: 'white',
        margin: {
            top: 10,
            right: 60,
            bottom: 10,
            left: 60
        },
        textTransform: 'uppercase',
    },
    searchValue: {
        height: 40,
        width: 1080,
        fontSize: 30,
        backgroundColor: 'rgba(0, 0, 0, 0.9)',
        margin: {
            top: 10,
            right: 60,
            bottom: 20,
            left: 60
        },
        color: 'white',
        'border-top': [2, 'solid', 'black'],
        'border-right': [2, 'solid', 'black'],
        'border-bottom': [2, 'solid', '#ff0055'],
        'border-left': [2, 'solid', 'black'],
    },
};
  
const SearchAreaDiv = ({ classes, value, handleChange, children }) => (
    <div className={classes.searchArea}>
        <h1 className={classes.searchTitle}>find your movie</h1>
        <input type="search" name="search-value" className={classes.searchValue} value={value} onChange={(e) => handleChange(e.target.value)} />
        { children }
    </div>
);

const StyledSearchAreaDiv = injectSheet(styles)(SearchAreaDiv);

export const SearchArea = (props: Props) => {

    const { value, handleChange, match, location } = props;

    const paramValue = match && match.params.value;

    return (
        <StyledSearchAreaDiv value={value} handleChange={handleChange} >
            <SearchFilter location={location} paramValue={paramValue} />
        </StyledSearchAreaDiv>
    );  
};

const mapDispatchToProps = dispatch => {
    return {
        handleChange: (value) => dispatch(saveInputValue(value)),
    };
};

const mapStateToProps = state => {
    return {
        value: state.filmListReducer.currentValue,
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(SearchArea);