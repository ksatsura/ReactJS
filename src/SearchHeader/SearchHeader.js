//imports
import React from 'react';
import SearchArea from '../SearchArea/SearchArea';
import CompanyTitle from '../CompanyTitle/CompanyTitle';
import { Route, Switch } from 'react-router-dom';
import '../style.css';

const SearchHeader = () => {

    return (
        <div className='header'>
            <Route path='/' component={ CompanyTitle } />
            <Switch>
                <Route exact path='/' render={() =>  <SearchArea /> } />
                <Route path='/search/Search :value' render={props =>  <SearchArea  location={props.location} match={props.match} /> } />
            </Switch>
        </div>
    );
};

export default SearchHeader;