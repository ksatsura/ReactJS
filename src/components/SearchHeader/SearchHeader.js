//imports
import React from 'react';
import SearchArea from '../SearchArea/SearchArea';
import CompanyTitle from '../CompanyTitle/CompanyTitle';
import '../../style.css';

const SearchHeader = (props) => {

    const { location, match } = props;

    return (
        <div className='header'>
            <CompanyTitle />
            <SearchArea location={location} match={match}/>
        </div>
    );
};

export default SearchHeader;