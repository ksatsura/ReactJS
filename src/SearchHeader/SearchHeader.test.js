import React from 'react';
import SearchHeader from './SearchHeader';
import SearchArea from '../SearchArea/SearchArea';
import CompanyTitle from '../CompanyTitle/CompanyTitle';
import { shallow } from 'enzyme';

describe('SearchHeader.js', () => {

    describe('renders', () => {

        it('renders for desktop', () => {
            const component = shallow(<SearchHeader />);

            expect(component).toMatchSnapshot();
            expect(component.find(SearchArea).length).toEqual(1);
            expect(component.find(CompanyTitle).length).toEqual(1);
        });

        it('renders for mobile', () => {
            const component = shallow(<SearchHeader isMobile />);

            expect(component).toMatchSnapshot();
            expect(component.find(SearchArea).length).toEqual(1);
            expect(component.find(CompanyTitle).length).toEqual(1);
        });
    });
});