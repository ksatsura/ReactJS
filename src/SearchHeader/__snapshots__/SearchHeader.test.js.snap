// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchHeader.js should render as expected 1`] = `
<div
  className="header"
>
  <Ð¡ompanyTitle />
  <Connect(SearchArea) />
</div>
`;
